Sources/SwiftDocC/LinkTargets/LinkDestinationSummary.swift

```diff
@@ -516,5 +574,90 @@ extension LinkDestinationSummary.Variant {
         usr = try container.decodeIfPresent(String?.self, forKey: .usr)
         declarationFragments = try container.decodeIfPresent(LinkDestinationSummary.DeclarationFragments?.self, forKey: .declarationFragments)
         taskGroups = try container.decodeIfPresent([LinkDestinationSummary.TaskGroup]?.self, forKey: .taskGroups)
+        topicImages = try container.decodeIfPresent([TopicImage]?.self, forKey: .topicImages)
+    }
+}
+
+// MARK: - Equatable conformance
+
+// Manually implement `==(_:_:)` since ``RenderReference`` doesn't conform to Equatable.
+// (Adding that conformance would add a Self requirement that would add the need for type erasure in other places)
+extension LinkDestinationSummary {
+    public static func == (lhs: LinkDestinationSummary, rhs: LinkDestinationSummary) -> Bool {
+        // The reference URL and usr have identity-like qualities.
+        // If those are different the other data-like properties would also be different.
+        guard lhs.referenceURL == rhs.referenceURL else { return false }
+        guard lhs.usr == rhs.usr else { return false }
+        
+        // Compare the data-like properties in the order that they are declared (above).
+        guard lhs.kind == rhs.kind else { return false }
+        guard lhs.language == rhs.language else { return false }
+        guard lhs.relativePresentationURL == rhs.relativePresentationURL else { return false }
+        guard lhs.title == rhs.title else { return false }
+        guard lhs.abstract == rhs.abstract else { return false }
+        guard lhs.availableLanguages == rhs.availableLanguages else { return false }
+        guard lhs.platforms == rhs.platforms else { return false }
+        guard lhs.taskGroups == rhs.taskGroups else { return false }
+        guard lhs.declarationFragments == rhs.declarationFragments else { return false }
+        guard lhs.redirects == rhs.redirects else { return false }
+        guard lhs.topicImages == rhs.topicImages else { return false }
+        guard lhs.variants == rhs.variants else { return false }
+        
+        // Since `RenderReference` doesn't conform to Equatable, the `references` need to be compared manually.
+        // There's a known number of render references types for the various `RenderReferenceType` cases.
+        guard lhs.references?.count == rhs.references?.count else { return false }
+        for (lhsRef, rhsRef) in zip(lhs.references ?? [], rhs.references ?? []) {
```

Comment:
Does the order of these matter for equatableness?

Reply:
In this implementation the order matters but [the initializers sort the references by their identifier][1] so in practice it doesn't matter. 

[1]: https://github.com/apple/swift-docc/pull/469/files/cb26ffea2edb32f75369daf3c7051f8f70928314#diff-a72178c5b706bcb7382fd00322f921a0b9ba03e844f63a8d5c11398c4b807fabR377-R379

```
let references = referenceIdentifiers
    .compactMap { renderNode.references[$0.identifier] }
    .sorted(by: \.identifier.identifier)
```

I'm of the opinion that two equal values should be indiscernible and that if order doesn't matter then the values should be modeled as unordered. I think we only expect each reference to occur once so we could use a `Set` instead of an `Array`.

I may still want to sort the references before encoding the references so that they are stable between builds but I can make that conditional based on `DOCC_JSON_PRETTYPRINT`.

Reply:
I tried to change this to a `Set` but couldn't because `RenderReference` doesn't conform to `Hashable`. Adding that conformance would add a `Self` requirement to the protocol which would mean that it needs type erasure everywhere which would be a big API breaking change.

I also tried to change this to a dictionary but I didn't feel that that made as much sense for the link destination summaries since the purpose of the `references` property is to list the collection of references in the content rather than to look up individual references.

Since neither of those alternatives worked well I feel that an array (with sorted values) is probably the best way to represent this.

Reply:
Makes sense. Thank you!